javascript/ql/src/semmle/javascript/security/dataflow/ZipSlipCustomizations.qll

```diff
@@ -119,6 +121,20 @@ module ZipSlip {
     BasenameSanitizer() { this = DataFlow::moduleImport("path").getAMemberCall("basename") }
   }
 
+  /**
+   * An expression that forces the output path to be in the current working folder.
+   * Recognizes the pattern: `path.join(cwd, path.join('/', orgPath))`.
+   */
+  class PathSanitizer extends Sanitizer, DataFlow::CallNode {
```

Comment:
Isn't this a general purpose path sanitizer or am I missing som subtlety?

Reply:
Yes, that is a general purpose path sanitizer.  

The same pattern is already supported by `js/path-injection`. But `js/path-injection` does the sanitization in multiple steps using flow-labels, so that is non-trivial to import without basically importing all of `js/path-injection` into `js/zip-slip`.

Reply:
Fine. 
Lets consider merging the two queries at another time.

Reply:
> Lets consider merging the two queries at another time.

I've opened https://github.com/github/codeql-javascript-team/issues/116 to track this